src/k2/dto/Expression.cpp

```diff
@@ -0,0 +1,298 @@
+#include "Expression.h"
+namespace k2 {
+namespace dto {
+namespace expression {
+
+struct SchematizedValue {
+    SchematizedValue(Value& v, SKVRecord& rec) : val(v), rec(rec), type(v.type) {
+        if (!val.fieldName.empty()) {
+            for (size_t i = 0; i < rec.schema->fields.size(); ++i) {
+                if (rec.schema->fields[i].name == val.fieldName) {
+                    sfieldIndex = i;
+                    nullLast = rec.schema->fields[i].nullLast;
+                    type = rec.schema->fields[i].type;
+                    break;
+                }
+            }
+        }
+    }
+    Value& val;
+    SKVRecord& rec;
+    FieldType type = FieldType::NOT_KNOWN;
+    bool nullLast = false;
+    int sfieldIndex = -1;
+
+    template <typename T>
+    std::optional<std::tuple<bool, T>> get() {
```

Comment:
Shouldn't this be std::tuple<bool, std::optional<T>> ?

Reply:
The api here matches along with the other question about what to do if a field isn't set.
If we believe that the correct model is to treat
- key is in schema but not set
- key is not in schema
the same, then this API has to return an empty optional in both cases, with nullLast set if non-null value was found